tests/templates/test_subroutines.py

```diff
@@ -1207,3 +1208,284 @@ def circuit():
             return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_wires)]
 
         assert np.allclose(circuit(), expectation)
+
+
+class TestPermute:
+    """Tests for the Permute template from the pennylane.templates.subroutine module."""
+
+    def test_invalid_inputs_qnodes(self):
+        """Tests if errors are thrown for invalid permutations with QNodes."""
+
+        dev = qml.device("default.qubit", wires=4)
+
+        with pytest.raises(ValueError):
```

Comment:
I would include the `match=` argument here, just to make sure the correct ValueError is being raised (since there are multiple different validations). This helps make the tests more atomic, and makes it easier for other developers to later quickly work out what test is testing what.